sdk/iot/hub/tests/cmocka/test_az_iot_hub_client.c

```diff
@@ -93,68 +93,76 @@ static void test_az_iot_hub_client_init_NULL_hub_hostname_id_fails(void** state)
   assert_precondition_checked(az_iot_hub_client_init(&client, test_device_id, AZ_SPAN_NULL, NULL));
 }
 
-static void test_az_iot_hub_client_user_name_get_NULL_client_fails(void** state)
```

Comment:
Nit: we may want to simplify the cmocka state/(void)state some time in the future. 2x less lines to scroll through for each test.

Reply:
Yea I agree. As it stands the compiler flags prevent us from doing it. One of the builds fails

Reply:
See provisioning tests: it requires a pragma where tests are added.